kasa/tapo/tapodevice.py

```diff
@@ -0,0 +1,160 @@
+"""Module for a TAPO device."""
+import base64
+import logging
+from datetime import datetime, timedelta, timezone
+from typing import Any, Dict, Optional, Set, cast
+
+from ..aesprotocol import TPLinkAes
+from ..credentials import Credentials
+from ..exceptions import AuthenticationException
+from ..smartdevice import SmartDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TapoDevice(SmartDevice):
+    """Base class to represent a TAPO device."""
+
+    def __init__(
+        self,
+        host: str,
+        *,
+        port: Optional[int] = None,
+        credentials: Optional[Credentials] = None,
+        timeout: Optional[int] = None,
+    ) -> None:
+        super().__init__(host, port=port, credentials=credentials, timeout=timeout)
+        self._state_information: Dict[str, Any] = {}
+        self._discovery_info: Optional[Dict[str, Any]] = None
+        self.protocol = TPLinkAes(host, credentials=credentials, timeout=timeout)
+
+    async def update(self, update_children: bool = True):
+        """Update the device."""
+        if self.credentials is None or self.credentials.username is None:
+            raise AuthenticationException("Tapo plug requires authentication.")
+
+        self._info = await self.protocol.query("get_device_info")
+        self._usage = await self.protocol.query("get_device_usage")
+        self._time = await self.protocol.query("get_device_time")
+
+        self._last_update = self._data = {
+            "info": self._info,
+            "usage": self._usage,
+            "time": self._time,
+        }
+
+        _LOGGER.debug("Got an update: %s", self._data)
+
+    @property
+    def sys_info(self) -> Dict[str, Any]:
+        """Returns the device info."""
+        return self._info
+
+    @property
+    def model(self) -> str:
+        """Returns the device model."""
+        return str(self._info.get("model"))
```

Comment:
```suggestion
        return self._info.get("model")
```
Is there a need to cast this? If not, please remove.

Reply:
The casting is needed to make mypy happy.  It seems consistent with properties in `SmartDevice`:
```
    def model(self) -> str:
        """Return device model."""
        sys_info = self._sys_info
        return str(sys_info["model"])
```